src/dia/Paper.mjs

```diff
@@ -2097,27 +2106,85 @@ export const Paper = View.extend({
+
+            const isTargetFormNode = this.FORM_CONTROL_TAG_NAMES.includes(target.tagName);
+
+            if (this.options.preventDefaultViewAction && !isTargetFormNode) {
+                // If the target is a form element, we do not want to prevent the default action.
+                // For example, we want to be able to select text in a text input or
+                // to be able to click on a checkbox.
+                evt.preventDefault();
+            }
+
+            if (isTargetFormNode) {
+                // If the target is a form element, we do not want to start dragging the element.
+                // For example, we want to be able to select text by dragging the mouse.
+                view.preventDefaultInteraction(evt);
+            }
+
+            const rootViewEl = view.el;
+
+            // Custom event
+            const eventNode = target.closest('[event]');
+            if (eventNode && rootViewEl !== eventNode && view.el.contains(eventNode)) {
+                const eventEvt = normalizeEvent($.Event(evt.originalEvent, {
+                    data: evt.data,
+                    // Originally the event listener was attached to the event element.
+                    currentTarget: eventNode
+                }));
+                this.onevent(eventEvt);
+                if (eventEvt.isDefaultPrevented()) {
+                    evt.preventDefault();
+                }
+                // `onevent` can stop propagation
+                if (eventEvt.isPropagationStopped()) return;
+                evt.data = eventEvt.data;
+            }
+
+            // Element magnet
+            const magnetNode = target.closest('[magnet]');
+            if (magnetNode && view.el !== magnetNode && view.el.contains(magnetNode)) {
+                const magnetEvt = normalizeEvent($.Event(evt.originalEvent, {
+                    data: evt.data,
+                    // Originally the event listener was attached to the magnet element.
+                    currentTarget: magnetNode
+                }));
+                this.onmagnet(magnetEvt);
+                if (magnetEvt.isDefaultPrevented()) {
+                    evt.preventDefault();
+                }
+                // `onmagnet` stops propagation when `addLinkFromMagnet` is allowed
+                if (magnetEvt.isPropagationStopped()) {
+                    // `magnet:pointermove` and `magnet:pointerup` events must be fired
+                    if (isContextMenu) return;
+                    this.delegateDragEvents(view, magnetEvt.data);
+                    return;
+                }
+                evt.data = magnetEvt.data;
+            }
+        }
+
+        if (isContextMenu) {
             this.contextMenuFired = true;
-            const contextmenuEvt = $.Event(evt, { type: 'contextmenu', data: evt.data });
+            const contextmenuEvt = $.Event(evt.originalEvent, { type: 'contextmenu', data: evt.data });
```

Comment:
breaking change?

Reply:
It was a bug. The `originalEvent` must point to a native event or undefined.

https://api.jquery.com/category/events/event-object/
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jquery/misc.d.ts#L4702